components/server/src/analytics.ts

```diff
@@ -38,10 +45,20 @@ export async function trackSignup(user: User, request: Request, analytics: IAnal
     });
 }
 
-function fullIdentify(user: User, request: Request, analytics: IAnalyticsWriter) {
-    //makes a full identify call for authenticated users
+async function fullIdentify(
+    user: User,
+    request: Request,
+    analytics: IAnalyticsWriter,
+    subscriptionService?: SubscriptionService,
```

Comment:
I think you can safely drop the optinal on `subscriptionService` since the only caller is from `trackLogic` which as the `subscriptionService` as required.

Reply:
```suggestion
    subscriptionService: SubscriptionService,
```

Reply:
This also means you can change a line below to 
```
const subscriptions = await subscriptionService.getNotYetCancelledSubscriptions(user, new Date().toISOString());
```

Reply:
woops, the latest build failed as we also use `fullIdentify` in `trackSignup` and we don't have access to the subscription service from the calling context there. reverting this change and squashing the commits then!